lib/TwoFactorAuth.php

```diff
@@ -78,10 +79,16 @@ public function getCode($secret, $time = null)
 
     /**
      * Check if the code is correct. This will accept codes starting from ($discrepancy * $period) sec ago to ($discrepancy * period) sec from now
+     * @param string   $secret
+     * @param string   $code This shouldn't be casted to integer - you may lose zeroes to the left
+     * @param int      $discrepancy
+     * @param int|null $time
+     * @param int      $timeslice
+     * @return bool
```

Comment:
I also added the `string $code` to a docblock to make it clear it won't work if casted - besides losing the zeroes, `hash_equals('123', 123)` returns `false`